File: abc/abc151/abc151_b.cs
// https://atcoder.jp/contests/abc151/submissions/9446921

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var K = input[1];
            var M = input[2];

            var A = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            if (A.Sum() + K < N * M)
            {
                Console.WriteLine("-1");
                return;
            }
            else if (A.Sum() >= N * M)
            {
                Console.WriteLine("0");
                return;
            }
            Console.WriteLine(N * M - A.Sum());
        }
    }
}

File: abc/abc151/abc151_b.fs
// https://atcoder.jp/contests/abc151/submissions/11937899

let [| N; K; M |] = stdin.ReadLine().Split() |> Array.map int

let As =
    stdin.ReadLine().Split()
    |> Array.map int
    |> Array.sum

match N * M - As with
| x when x > K -> -1
| x when x > 0 -> x
| _ -> 0
|> stdout.WriteLine
